logging/google/cloud/logging/handlers/_helpers.py

```diff
@@ -69,6 +79,33 @@ def get_trace_id_from_flask():
     return trace_id
 
 
+def get_trace_id_from_webapp2():
+    """Get trace_id from webapp2 request headers.
+
+    :rtype: str
+    :return: Trace_id in HTTP request headers.
+    """
+    if webapp2 is None:
+        return None
+
+    try:
+        # get_request() succeeds if we're in the middle of a webapp2
+        # request, or raises an assertion error otherwise:
+        # "Request global variable is not set".
+        req = webapp2.get_request()
+    except AssertionError:
+        return None
+
+    header = req.headers.get(_WEBAPP2_TRACE_HEADER)
+
+    if header is None:
+        return None
+
+    trace_id = header.split('/', 1)[0]
```

Comment:
It's only one line of code, so I was happy copying it, but I can make a helper function too if you think it's clearer.